### PR TITLE
doc(reference): Standardize docstring syntax

### LINKED ISSUE #334: Normalize use of text styles in docstrings
TBD, but current thoughts:

* For types, use double-backticks
* For referencing methods or params, use single-backticks

### PR DESCRIPTION
**Pull request by kgriffs**
Unify the use of markup in docstrings, particulary relating to the use of
backticks and asterisks. Also clean up any remaining minor inconsistencies or
errors in the docstrings.

Closes #334

### CODE REVIEW

File: falcon/request_helpers.py [hunk 1]
```diff
@@ -35,20 +35,20 @@ def fget(self):
...
     the request's content length is passed to the method.
 
-    This class normalizes wsgi.input behavior between WSGI servers
+    This class normalizes *wsgi.input* behavior between WSGI servers
     by implementing non-blocking behavior for the cases mentioned
     above.
 
     Args:
-        stream: Instance of socket._fileobject from environ['wsgi.input']
+        stream: Instance of ``socket._fileobject`` from ``environ['wsgi.input']``
```
Comment (rsyed83) [inline falcon/request_helpers.py]: Line too long

File: falcon/testing/resource.py [hunk 2]
```diff
@@ -30,13 +30,13 @@ class TestResource:
...
+        req (falcon.Request): Request object passed into the `on_get`
             responder
-        resp (falcon.Response): Response object passed into the *on_get*
+        resp (falcon.Response): Response object passed into the `on_get`
             responder
-        kwargs (dict): Keyword arguments passed into the *on_get*
+        kwargs (dict): Keyword arguments passed into the `on_get`
             responder, if any
-        called (bool): True if *on_get* was ever called; False otherwise
+        called (bool): ``True`` if `on_get` was ever called; ``False`` otherwise
```
Comment (rsyed83) [inline falcon/testing/resource.py]: Line too long

File: falcon/response.py [hunk 3]
```diff
@@ -188,17 +188,17 @@ def set_headers(self, headers):
 
         Args:
             headers (dict or list): A dictionary of header names and values
-                to set, or list of (name, value) tuples. Both names and
-                values must be of type str or StringType, and only character
-                values 0x00 through 0xFF may be used on platforms that use
-                wide characters.
+                to set, or ``list`` of (*name*, *value*) tuples. Both *name* and
```
Comment (rsyed83) [inline falcon/response.py]: Line too long

File: falcon/request.py [hunk 4]
```diff
@@ -637,24 +638,24 @@ def get_param_as_int(self, name,
...
 
         Args:
-            name (str): Parameter name, case-sensitive (e.g., 'limit')
-            required (bool, optional): Set to True to raise HTTPBadRequest
-                instead of returning gracefully when the parameter is not
-                found or is not an integer (default False).
+            name (str): Parameter name, case-sensitive (e.g., 'limit').
+            required (bool, optional): Set to ``True`` to raise
+                ``HTTPBadRequest`` instead of returning ``None`` when the
+                parameter is not found or is not an integer (default ``False``).
```
Comment (rsyed83) [inline falcon/request.py]: Line too long

File: falcon/request.py [hunk 5]
```diff
@@ -637,24 +638,24 @@ def get_param_as_int(self, name,
...
             min (int, optional): Set to the minimum value allowed for this
                 param. If the param is found and it is less than min, an
-                HTTPError is raised.
+                ``HTTPError`` is raised.
             max (int, optional): Set to the maximum value allowed for this
                 param. If the param is found and its value is greater than
-                max, an HTTPError is raised.
-            store (dict, optional): A dict-like object in which to place the
+                max, an ``HTTPError`` is raised.
+            store (dict, optional): A ``dict``-like object in which to place the
```
Comment (rsyed83) [inline falcon/request.py]: Line too long

File: falcon/request.py [hunk 6]
```diff
@@ -701,27 +702,31 @@ def get_param_as_bool(self, name, required=False, store=None,
...
-            required (bool, optional): Set to True to raise HTTPBadRequest
-                instead of returning gracefully when the parameter is not
-                found or is not a recognized bool-ish string (default False).
-            store (dict, optional): A dict-like object in which to place the
+            name (str): Parameter name, case-sensitive (e.g., 'detailed').
+            required (bool, optional): Set to ``True`` to raise
+                ``HTTPBadRequest`` instead of returning ``None`` when the
+                parameter is not found or is not a recognized boolean
+                string (default ``False``).
+            store (dict, optional): A ``dict``-like object in which to place the
```
Comment (rsyed83) [inline falcon/request.py]: Line too long

File: falcon/request.py [hunk 7]
```diff
@@ -766,22 +771,22 @@ def get_param_as_list(self, name,
...
+                will be returned. For example, passing ``int`` will
                 transform list items into numbers.
-            required (bool, optional): Set to True to raise HTTPBadRequest
-                instead of returning gracefully when the parameter is not
-                found or is not an integer (default False)
-            store (dict, optional): A dict-like object in which to place the
+            required (bool, optional): Set to ``True`` to raise
+                ``HTTPBadRequest`` instead of returning ``None`` when the
+                parameter is not found (default ``False``).
+            store (dict, optional): A ``dict``-like object in which to place the
```
Comment (rsyed83) [inline falcon/request.py]: Line too long

**Comment by rsyed83** [comment]
Just realized that after seeing the pep8 failures you likely could've identified the lines locally.  Sorry for the spam!

**Comment by kgriffs** [comment]
@rsyed83 no worries, thanks for the review!

**Pull request merged**